### PR TITLE
AF-1863 dart1 and dart 2 compat

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Overview
Updates dependencies and CI to work under Dart 2 and Dart 1. No code changes are needed to have it work under Dart 2.

# Testing
 - [ ] CI passes under Dart 1 and Dart 2

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-  - pub run dart_dev analyze
-  - pub run dart_dev test
-  - pub run dart_dev coverage --no-html
-  - bash <(curl -s https://codecov.io/bash) -f coverage/coverage.lcov
+  - 1.24.3
+  - dev
+dart_task:
+  - test: --platform vm
+  - test: --platform chrome
+  - dartanalyzer: lib example
```
Comment (evanweible-wf) [inline .travis.yml]: any reason not to also analyze tool and test directories? And if not, I think you can just leave this as `- dartanalyzer`

File: pubspec.yaml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 dev_dependencies:
-  dartdoc: '>=0.8.0 <=0.10.0'
-  dart_style: '>=0.1.8 <0.3.0'
-  coverage: ^0.7.9
-  browser: any
-  dart_dev: ^1.7.2
+  build_runner: ">=0.6.0 <1.0.0"
+  build_test: ">=0.9.0 <1.0.0"
+  build_web_compilers: ">=0.2.0 <1.0.0"
+  coverage: ">=0.7.9"
```
Comment (evanweible-wf) [inline pubspec.yaml]: does `pub publish --dry-run` complain about not having an upper bound? I've been using `<0.13.0`

File: .travis.yml (hunk 1 above)
Comment (robbecker-wf) [reply .travis.yml]: no reason .. just cause copy/paste. will adjust.

File: pubspec.yaml (hunk 2 above)
Comment (robbecker-wf) [reply pubspec.yaml]: it does not complain, but I suppose we should have one. will update.

**Comment by maxwellpeterson-wf** [comment]
@Workiva/release-management-pp

**Pull request merged**